tests/regression/nextcloud-rule-exclusions-plugin/9508993.yaml

```diff
@@ -0,0 +1,29 @@
+---
+meta:
+  author: "Jean-Kevin Kpadey"
+  description: "Nextcloud Rule Exclusions Plugin"
+  enabled: true
+  name: 9508993.yaml
+tests:
+  - test_title: 9508993-1
+    desc: |
+      Moving cards on Deck apps.
+      Target: description
+    stages:
+      - stage:
+          input:
+            dest_addr: 127.0.0.1
+            headers:
+              Host: localhost
+              User-Agent: OWASP ModSecurity Core Rule Set
+              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
+            port: 80
+            method: PUT
+            uri: /apps/deck/cards/1/reorder
+            data: |
+              description=This is a description.
+
+              Use this [link](https://github.com/microsoft/vscode/issues?page=3&q=is%3Aissue+is%3Aopen).
+          output:
+            no_log_contains: |
+              id "911100"|id "932200"|id "933210"|id "942200"|id "942260"|id "942370"|id "942430"|id "942440"|id "949110"|id "980130"
```

Comment:
@theseion 949110 and 980130 are for anomoly scoring and corelations respectively, is it worth it to test for these rules, at least for 949110 it should never trigger when the regression tests are running?

Reply:
That's actually an interesting observation. I've been wandering recently about false positive tests. Negative matches are obviously a bit tricky, but we could just say: no rule should trigger. This is probably the case in 99% of the cases (at least for exclusion rule plugins). On the other hand, it could also mean that many test start failing at the same time, e.g., when a new rule is introduced that triggers on something.
Still, seeing the failure is probably more important. What do you think, should we discuss replacing individual rule matches with "no rules should trigger" for the Nextcloud plugin?

Reply:
I've been wondering about that as well, Nextcloud isn't very CRS friendly and this will really make it easier to ensure there aren't any FPs with Nextcloud. It'll be much easier to maintain in the long run and cause fewer issues for end users. But I'm not sure how that would work, would you just write a regex rule that checks for any rule ID within the 911000-959999 range?
But for this test in particular, should we remove 949110 and 980130?

Reply:
I think I'd do `no_log_contains: id "\d+"`. That would match __any__ rule. But yes, let's postpone that. I agree, let's remove those two matches to be consistent.